### PR TITLE
[cirqflow] QuantumExecutableGroup

### PR DESCRIPTION
**Pull request by mpharrigan**
This is a simple container for a sequence of quantum executables. It ensures immutability and caches potentially expensive hash values.
RequestResponse2 (1)

follows #4527

### CODE REVIEW

File: cirq-google/cirq_google/workflow/quantum_executable.py [hunk 1]
```diff
@@ -157,3 +157,53 @@ def __repr__(self):
...
+
+        return f'QuantumExecutable(executables=[{exe_str}])'
+
+    def __repr__(self):
+        return _compat.dataclass_repr(self, namespace='cirq_google')
+
+    def __hash__(self):
+        return self._hash
+
+    def _json_dict_(self):
```
Comment (MichaelBroughton) [inline cirq-google/cirq_google/workflow/quantum_executable.py:208]: Nit: missing some return type hints here.

File: cirq-google/cirq_google/workflow/quantum_executable_test.py [hunk 2]
```diff
@@ -117,3 +122,51 @@ def test_quantum_executable_inputs():
...
+    exes1 = list(_get_quantum_executables())
+    exes2 = tuple(_get_quantum_executables())
+
+    eg1 = QuantumExecutableGroup(exes1)
+    eg2 = QuantumExecutableGroup(exes2)
+    assert hash(eg1) == hash(eg2)
+    assert eg1 == eg2
+
+
+def test_quantum_executable_group_container(tmpdir):
```
Comment (MichaelBroughton) [inline cirq-google/cirq_google/workflow/quantum_executable_test.py:149]: Nit: Can you split this out so \_\_len__, \_\_iter__, \_\_repr__ and \_\_str__ are tested in one test and the json serialization deserialization consistency is tested in another ?

**Review by MichaelBroughton — APPROVED** [review]
LGTM with some minor nits.

File: cirq-google/cirq_google/workflow/quantum_executable.py (hunk 1 above)
Comment (mpharrigan) [reply cirq-google/cirq_google/workflow/quantum_executable.py:208]: added

File: cirq-google/cirq_google/workflow/quantum_executable_test.py (hunk 2 above)
Comment (mpharrigan) [reply cirq-google/cirq_google/workflow/quantum_executable_test.py]: I can and I have

**Comment by mpharrigan** [comment]
@MichaelBroughton PTAL

**Review by MichaelBroughton — APPROVED** [review]
LGTM. Feel free to merge after CI fix.

File: cirq-google/cirq_google/workflow/quantum_executable_test.py (hunk 2 above)
Comment (MichaelBroughton) [reply cirq-google/cirq_google/workflow/quantum_executable_test.py]: 😎

**Comment by mpharrigan** [comment]
do you know why my disable=line-too-long are no longer working? It was fine before I split up the tests

**Pull request merged**